crates/mem-pool/src/sync/mq/gw_kafka.rs

```diff
@@ -0,0 +1,130 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use gw_types::bytes::Bytes;
+use gw_types::packed::{RefreshMemBlockMessage, RefreshMemBlockMessageUnion};
+use gw_types::prelude::{Builder, Entity, Reader};
+use kafka::client::{Compression, FetchOffset, GroupOffsetStorage};
+use kafka::producer::{AsBytes, Record};
+
+use crate::sync::fan_in::FanInMemBlock;
+
+use super::{Consume, Produce};
+
+pub(crate) struct Producer {
+    producer: kafka::producer::Producer,
+    topic: String,
+}
+
+impl Producer {
+    pub(crate) fn new(hosts: Vec<String>, topic: String) -> Result<Self> {
+        let mut client = kafka::client::KafkaClient::new(hosts);
+        client.load_metadata(&[&topic]).map_err(|err| {
```

Comment:
Rename this function to connect. The name new implies the function shouldn't has any side-effect